pyhive/sqlalchemy_hive.py

```diff
@@ -517,6 +518,37 @@ def _check_unicode_description(self, connection):
         # We decode everything as UTF-8
         return True
 
+
+from sqlalchemy.ext.compiler import compiles
+from sqlalchemy.sql.expression import Select
+
+
+@compiles(Select)
+def select_star_if_all_columns(select, compiler, **kwargs):
+    froms = select.froms
+    select_columns = list(select.columns.keys())
+    from_columns = list(chain.from_iterable((f.columns.keys() for f in froms)))
+
+    # hive special cases star, so if we want the first N rows quickly, we
+    # need to pass it a star. If we let sqlalchemy have its way, then every
+    # column is explicitly named and mapred is triggered.
+    if select_columns == from_columns:
+        select = Select(columns=['*'],
```

Comment:
This seems kinda fragile. If we can whitelist to certain query types as mentioned above then this can be simplified.